### PR TITLE
Support visualizing text grid files with Praat.

### PR DESCRIPTION
**Pull request by csukuangfj**
Example results:

Screen Shot 2021-06-07 at 7 54 04 PM

[all.pdf](https://github.com/k2-fsa/snowfall/files/6608583/all.pdf)
[a-0.5-3.5.pdf](https://github.com/k2-fsa/snowfall/files/6608586/a-0.5-3.5.pdf)

### CODE REVIEW

**Comment by csukuangfj** [comment]
See https://github.com/k2-fsa/snowfall/blob/cf5ba85c5d10066eed2008716bc27a24801ae00f/test/fixtures/README.rst
for how to run it:

```bash
$ snowfall ali visualize -i ./a.flac -t a.TextGrid -w 20 -s 0.5 -e 3.5 -o a-0.5-3.5.pdf
$ snowfall ali visualize -i ./a.flac -t a.TextGrid -w 100 -o all.pdf
```

---

Praat can be installed by visiting https://www.fon.hum.uva.nl/praat/

Screen Shot 2021-06-07 at 7 56 42 PM

File: snowfall/tools/ali.py [hunk 1]
```diff
@@ -105,3 +109,77 @@ def compute_edit_distance(ref_ali: Dict[str, Alignment],
...
+      Save as PDF file: "{Path(output_file).resolve()}"
+    '''
+    tmp_file = tempfile.NamedTemporaryFile(delete=False)
+    tmp_file.close()
+    tmp_name = tmp_file.name
+    with open(tmp_name, 'w') as f:
+        f.write(command)
+
+    cmd = f'{p} --run {tmp_name}'
+    ret = os.system(f'{p} --run {tmp_name}')
```
Comment (pzelasko) [inline snowfall/tools/ali.py:181]:
```suggestion
    ret = os.system(cmd)
```

**Review by pzelasko — COMMENTED** [review]
That's pretty cool! Are you also planning to add a utility for storing alignments as textgrid files?

**Comment by csukuangfj** [comment]
> That's pretty cool! Are you also planning to add a utility for storing alignments as textgrid files?

I think we can add several utilities: one to convert snowfall alignments to ctm, another to text grid.

The above example contains only one tier, phones. We can also add another tier, words. And they can be visualized together in a single figure.

File: snowfall/tools/ali.py (hunk 1 above)
Comment (csukuangfj) [reply snowfall/tools/ali.py]: Thanks! Fixed.

**Comment by danpovey** [comment]
Great!!  Merge when you think it makes sense...

**Pull request merged**